src/main/java/com/commercetools/pspadapter/paymentHandler/impl/PaymentHandler.java

```diff
@@ -99,32 +115,69 @@ public PaymentHandleResponse createPayment(@Nonnull String ctpPaymentId) {
-
-                                return paypalPlusFacade.getPaymentService().create(paypalPlusPayment)
-                                        .thenCompose(createdPpPayment -> setApprovalUrlAndInterfaceId(createdPpPayment, ctpPaymentId))
+                                AddInterfaceInteraction action = createAddInterfaceInteractionAction(paypalPlusPayment, REQUEST);
+                                // 1. save create paypal request to ctp payment's interface interaction
+                                return ctpFacade.getPaymentService().updatePayment(ctpPayment, Collections.singletonList(action))
+                                        .thenCompose(payment -> {
+                                            // 2. create paypal payment
+                                            return paypalPlusFacade.getPaymentService().create(paypalPlusPayment)
+                                                    .thenCompose(createdPpPayment -> {
+                                                                List<UpdateAction<io.sphere.sdk.payments.Payment>> updateActions
+                                                                        = new ArrayList<>(getApprovalUrlAndInterfaceIdAction(createdPpPayment));
+                                                                AddInterfaceInteraction interaction = createAddInterfaceInteractionAction(createdPpPayment, RESPONSE);
+                                                                updateActions.add(interaction);
+                                                                // 3. save approval url and payment ID and response from paypal to ctp payment
+                                                                return ctpFacade.getPaymentService().updatePayment(payment, updateActions)
+                                                                        .thenApply(ignore -> {
+                                                                            return createdPpPayment;
+                                                                        });
+                                                            }
+                                                    );
+                                        })
+                                        // 4. return approval url
                                         .thenApply(PaymentMapper::getApprovalUrl)
-                                        .thenApply(approvalUrlOpt -> approvalUrlOpt
-                                                .map(PaymentHandleResponse::of201CreatedApprovalUrl)
-                                                .orElseGet(() -> of400BadRequest(
-                                                        format("Payment or cart for ctpPaymentId=[%s] not found", ctpPaymentId))));
+                                        .thenApply(approvalUrlOpt -> {
+                                            return approvalUrlOpt
```

Comment:
braces and return not needed.

Reply:
https://github.com/commercetools/commercetools-paypal-plus-integration/pull/49/commits/962bc46ebad0dad1c761d7109add5edaf4cf7449